src/content/subgames.cpp

```diff
@@ -142,46 +158,42 @@ SubgameSpec findSubgame(const std::string &id)
 {
 	if (id.empty())
 		return SubgameSpec();
-	std::string share = porting::path_share;
-	std::string user = porting::path_user;
-
-	// Get games install locations
-	Strfnd search_paths(getSubgamePathEnv());
 
-	// Get all possible paths fo game
-	std::vector<GameFindPath> find_paths;
-	while (!search_paths.at_end()) {
-		std::string path = search_paths.next(PATH_DELIM);
-		path.append(DIR_DELIM).append(id);
-		find_paths.emplace_back(path, false);
-		path.append("_game");
-		find_paths.emplace_back(path, false);
+	std::string idv = normalizeGameId(id);
+
+	GamePathMap gamepaths = getAvailableGamePaths();
+	auto found = gamepaths.find(idv);
+	if (found == gamepaths.end()) { // Failed to find the game, try to find aliased game
+		[&](){
```

Comment:
Why the IIFE? `goto` for breaking out of nested loops is fine. Sadly C++ doesn't let us label loops to break out of them.

Reply:
I think it was you who told me to use a lambda instead of a goto :upside_down_face:

Reply:
Implemented anyway